students/roslyn_m/lesson09/mailroom_lesson9.py

```diff
@@ -0,0 +1,205 @@
+
+    @property
+    def average(self):
+        """ Calculates average donation amount
+                        :return: average donation
+                        """
+        return sum(self.donations)/len(self.donations)
+
+    @property
+    def qty(self):
+        """ Adds amount of donations.
+                        :return: qty of donations
+                        """
+        return len(self.donations)
+
+    @property
+    def total(self):
+        """ totals all donations
+                        :return: sum donations
+                        """
+        return sum(self.donations)
+
+    def __str__(self):
+        """ user friendly string describing object
+                        :return: (string)
+                        """
+        return "Name: {self.first} {self.last}, Donations: {self.donations}".format(self=self)
+
+# -------------------- IO Classes ---------------------- #
+
+
+class IO():
+
+    @staticmethod
+    def options_menu():
+        """ Show menu of options to user.
+                :return: nothing
+                """
+        print(
+            "\nMenu of actions:" + '\n \t1) Send a Thank You \n \t2) Create a report \n \t3) Create Thank You files for all donors \n \t4) Quit\n')
+
+    @staticmethod
+    def option_input():
+        """ Gather input from user on what action they want.
+                    :return: user_selection (string)
+                    """
+        while True:
+            user_selection = input('Please select option 1-3: ')
+            try:
+                user_selection = int(user_selection)
+                break
+            except ValueError:
+                print("You must enter a number from 1-4")
+        return user_selection
+
+    @staticmethod
+    def person_input():
+        """ Gather input from user on donor name.
+                :return: user_input (string)
+                """
+        user_input = input(
+            'Please type the FULL NAME of the donor who you would like to write a thank you note to. If you would like to see a complete donor report, type "report": ')
+        user_input = user_input.title()
+        return user_input
+
+    @staticmethod
+    def donation_input():
+        """ Gather input from user on donation amount.
+            :return: donation_amt (float)
+            """
+        try:
+            donation_amt = float(input("Please enter the donation amount: "))
+        except ValueError:
+            print("Your input was not valid!")
+            donation_amt = float(input("Please enter the donation amount: "))
+        if donation_amt < 0:
+            donation_amt = float(input("Please enter a donation amount greater than 0: "))
+        return donation_amt
+
+    @staticmethod
+    def print_to_user(string):
+        """ print data to user
+            return: nothing
+                """
+        print(string)
+
+
+# ---------------- Processing Classes ------------------ #
+
+class Processing():
```

Comment:
I believe this should be a DonorCollection class that holds reference to multiple donor objects and below functions should be class methods instead of staticmethods.

Notice that they all accept donor_all param - which usually is an indicator that it should be part of the class instance.

You can review the mailroom guidelines for more details.